text/0110-periodic-full-compaction.md

```diff
@@ -0,0 +1,317 @@
+# Periodic Full Compaction
+
+Author: [Alex Feinberg](https://github.com/afeinberg)
+
+Reviewers: [Connor](https://github.com/Connor1996),
+[Tony](https://github.com/tonyxuqqi), [Andy](https://github.com/v01dstar),
+others
+
+## Introduction
+
+**Periodic full compaction** is a scheduled task that starts at specified times
+of the day on each TiKV node and compacts the column families of all regions *at
+all levels including the bottommost (L6)*. In order to reduce to impact running
+such a heavy-weight task would have on the cluster's ability to serve traffic,
+full compaction is incremental: before the next range (presently a region) is
+compacted, we check if the load is below a certain threshold; if the threshold
+is exceeded, we pause the task until the load is again below a certain
+threshold.
+
+## Motivation
```

Comment:
Maybe we can mention that tikv-ctl compact-cluster may impact online user traffic a lot, makes it often not viable for product environment usage without downtime.  This tool provides the way to achieve the same goal, but in a more controllable way